src/v/archival/arrow_writer.cc

```diff
@@ -0,0 +1,323 @@
+#include "bytes/iobuf.h"
+#include "bytes/iostream.h"
+#include "cloud_storage/remote.h"
+#include "cloud_storage/types.h"
+#include "cluster/partition.h"
+#include "model/fundamental.h"
+#include "model/record.h"
+
+#include <seastar/core/loop.hh>
+#include <seastar/util/file.hh>
+
+#include <archival/arrow_writer.h>
+#include <arrow/scalar.h>
+#include <arrow/type_fwd.h>
+#include <arrow/visitor.h>
+
+#include <filesystem>
+#include <memory>
+#include <string_view>
+
+ss::future<bool> datalake::write_parquet(
+  const std::filesystem::path inner_path,
+  ss::shared_ptr<storage::log> log,
+  model::offset starting_offset,
+  model::offset ending_offset) {
+    storage::log_reader_config reader_cfg(
+      starting_offset,
+      ending_offset,
+      0,
+      10ll * 1024ll * 1024ll
+        * 1024ll, // FIXME(jcipar): 10 GiB is probably too much.
```

Comment:
Check out the operators in `base/units.h` for more convenient access to `10_GiB` :)